### PR TITLE
feat(option): make layer interface to implement logic state store

### PR DESCRIPTION
**Pull request by rizalfadlila**
hello @dlsrb6342, please kindly review my pr. 

make layer interface to implement logic state store to support implement store with another store (mongodb, redis, etc)

thank you

### CODE REVIEW

**Comment by rizalfadlila** [comment]
Hi @mingrammer please kindly my pr. i need used to mongodb to state store. Thank you

**Comment by rizalfadlila** [comment]
Hi @dlsrb6342, can you help me to review this pr? i need use to kinesumer as soon as possibel.

**Comment by mingrammer** [comment]
I'll review it today. Sorry for late reply.

File: statestore.go [hunk 1]
```diff
@@ -16,19 +16,33 @@ var (
 	ErrEmptyShardIDs = errors.New("kinesumer: empty shard ids given")
 )
 
-type db struct {
-	client *dynamo.DB
-	table  *dynamo.Table
-}
+type (
+	StoreRepositories interface {
```
Comment (mingrammer) [inline statestore.go:20]: Please rename it to `StateStore`!

Comment (rizalfadlila) [reply statestore.go]: fixed @mingrammer

**Review by mingrammer — APPROVED** [review]
LGTM. Thank you for contributing!

**Comment by mingrammer** [comment]
v0.2.3 is now released.

**Pull request merged**